scripts/update_docs_toc.py

```diff
@@ -0,0 +1,79 @@
+# /// script
+# requires-python = ">=3.13"
+# dependencies = ["markdown~=3.8", "mkdocs~=1.6.1"]
+# ///
+
+"""Update `docs/README.md`'s TOC."""
```

Comment:
```suggestion
"""Update `docs/README.md`'s table of contents.

This script generates a list of links to level-2 headers and writes that back
to `docs/README.md`. It is run in CI via `autogenerate_files.sh`.
"""
```

Reply:
It would also be great if you could add a short docstring to each function in this file. It doesn't have to be more than a sentence or two. We won't be looking at this code much, so it'll help refresh our memory about how it works when we come back to it at a later date if it's well commented.